prosper/datareader/intrinio/auth.py

```diff
@@ -0,0 +1,104 @@
+"""prosper.datareader.intrinio.auth: handle authentication/validation"""
+
+import requests
+
+from .. import exceptions
+from .. import config
+
+BASE_URL = 'https://api.intrinio.com/'
+class IntrinioHelper(object):
+    """parent class for handling requests to Intrininio
+
+    Notes:
+        See https://intrinio.com/account for account keys
+
+    Args:
+        username (str): username for direct-auth
+        password (str): password for direct-auth
+        public_key (str): API key for indirect access
+
+    Raises:
+        InvalidAuth: will not be able to access Intrinio feeds
+
+    """
+
+    def __init__(self, username='', password='', public_key='',):
+        self.__user = username
+        self.__password = password
+        self.__public_key = public_key
+
+        if not bool(self):
```

Comment:
REVIEW ME: is __bool__ the best tool to use in this case? __bool__ should never return False b/c of raise